src/exchanges/fetch.ts

```diff
@@ -29,7 +29,7 @@ export const fetchExchange: Exchange = ({ forward }) => {
           filter(op => op.operationName === 'teardown' && op.key === key)
         );
 
-        return pipe(createFetchSource(operation), takeUntil(teardown$));
+        return pipe(createFetchSource(operation, operation.operationName === 'query' && client.preferGetMethod), takeUntil(teardown$));
```

Comment:
We shouldn’t ever have to read from the client directly. Let’s move this to the operation context.

The reason for that has generally always been that the client’s option are opaque and separate from the operations and how they’re created and are dealt with in the exchange pipeline. So any option on the context can be seen, set, and manipulated by other exchanges. The same doesn’t go for the client.

To be fair, we should probably type the `Client` on the exchange using an interface that only contains the method we’d like to be exposed on the exchanges.